app/components/primer/beta/avatar.rb

```diff
@@ -27,38 +27,154 @@ class Avatar < Primer::Component
+      # @param unique_id [String, Integer] Optional unique identifier for generating consistent avatar colors across renders.
       # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
-      def initialize(src:, alt: nil, size: DEFAULT_SIZE, shape: DEFAULT_SHAPE, href: nil, **system_arguments)
+      def initialize(src: nil, alt: nil, size: DEFAULT_SIZE, shape: DEFAULT_SHAPE, href: nil, unique_id: nil, **system_arguments)
         @href = href
+        @unique_id = unique_id
+        @shape = fetch_or_fallback(SHAPE_OPTIONS, shape, DEFAULT_SHAPE)
+        @alt = alt
+        @size = fetch_or_fallback(SIZE_OPTIONS, size, DEFAULT_SIZE)
+
+        # Validate that either src or alt is provided
+        validate_src_or_alt(src, alt)
+
         @system_arguments = deny_tag_argument(**system_arguments)
         @system_arguments[:tag] = :img
-        @system_arguments[:src] = src
         @system_arguments[:alt] = alt
-        @system_arguments[:size] = fetch_or_fallback(SIZE_OPTIONS, size, DEFAULT_SIZE)
-        @system_arguments[:height] = @system_arguments[:size]
-        @system_arguments[:width] = @system_arguments[:size]
+        @system_arguments[:size] = @size
+        @system_arguments[:height] = @size
+        @system_arguments[:width] = @size
+
+        # Generate SVG fallback if src is nil/blank (must happen after @size is set)
+        @system_arguments[:src] = src.presence || generate_fallback_svg
 
         @system_arguments[:classes] = class_names(
           system_arguments[:classes],
           "avatar",
-          "avatar-small" => size < SMALL_THRESHOLD,
-          "circle" => shape == DEFAULT_SHAPE,
+          "avatar-small" => @size < SMALL_THRESHOLD,
+          "circle" => @shape == DEFAULT_SHAPE,
           "lh-0" => href # Addresses an overflow issue with linked avatars
         )
       end
 
       def call
-        if @href
-          render(Primer::Beta::Link.new(href: @href, classes: @system_arguments[:classes])) do
-            render(Primer::BaseComponent.new(**@system_arguments.except(:classes))) { content }
+        avatar_element =
+          if @href
+            render(Primer::Beta::Link.new(href: @href, classes: @system_arguments[:classes])) do
+              render(Primer::BaseComponent.new(**@system_arguments.except(:classes))) { content }
+            end
+          else
+            render(Primer::BaseComponent.new(**@system_arguments)) { content }
           end
+
+        # Wrap the avatar in an <avatar-fallback> custom element if the source
+        # is an SVG data URI (for client-side color correction)
+        if @system_arguments[:src]&.start_with?("data:image/svg+xml")
+          render(Primer::BaseComponent.new(
+            tag: :"avatar-fallback",
+            data: {
+              unique_id: @unique_id,
+              alt_text: @alt
+            }
+          )) { avatar_element }
+        else
+          avatar_element
+        end
```

Comment:
There's a component for exactly this: `Primer::ConditionalWrapper`:

```suggestion
        render(Primer::ConditionalWrapper.new(condition: @system_arguments[:src]&.start_with?("data:image/svg+xml"), tag: :"avatar-fallback", data: { unique_id: @unique_id, alt_text: @alt }) do
          avatar_element
        end
```